azavea/raster-foundry
# Local upload UI

## Overview

This PR adds the UI necessary to complete local uploading of geotiffs.

From any datasource's detail page, the user can being an 'import' and select files from their file system to upload. A new upload will be created in the database and the files will be streamed to S3. After the upload is complete, the status of the upload in the DB is set to `UPLOADED` to allow airflow to detect and begin the actual import process.

We filter out any file that is not of extension `.tif`.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

<img width="848" alt="screen shot 2017-04-04 at 4 31 30 pm" src="https://cloud.githubusercontent.com/assets/2442245/24677569/410179a6-1955-11e7-8d5f-e96b22f91618.png">
<img width="854" alt="screen shot 2017-04-04 at 4 31 37 pm" src="https://cloud.githubusercontent.com/assets/2442245/24677570/410166b4-1955-11e7-8d65-3296a7217149.png">
<img width="862" alt="screen shot 2017-04-04 at 4 33 59 pm" src="https://cloud.githubusercontent.com/assets/2442245/24677576/4428677a-1955-11e7-8065-9e70647740eb.png">
<img width="850" alt="screen shot 2017-04-04 at 4 34 12 pm" src="https://cloud.githubusercontent.com/assets/2442245/24677577/442b1e02-1955-11e7-8353-a2a795aa7495.png">
<img width="837" alt="screen shot 2017-04-04 at 4 38 14 pm" src="https://cloud.githubusercontent.com/assets/2442245/24677578/442ccf68-1955-11e7-907c-eabe39815257.png">


### Notes

This PR includes a non-npm vendor file (the aws-sdk for js, custom built with only the necessary S3 module (otherwise the file is huge). This isn't possible to obtain via npm, so I turned it into a module. There is probably a better way to handle this, but it works.

## Testing Instructions

 * Go to the 'Imports' section from the nav-bar, and view your datasources
 * Select a datasource and hit the 'Start an Import' button
 * Go through the process and check all sorts of things like not being able to go to the next step without files selected through various means
 * Check your db after uploads are completed and see that a row has been added to the `uploads` table each time you go through the process
 * Check S3, and see that within the development data bucket, there exists a `user_uploads -> <user_id> -> <upload_id>` folder with your uploaded files in it

Connects #1267
